reframe/core/schedulers/__init__.py

```diff
@@ -270,8 +270,10 @@ def guess_num_tasks(self):
                           len(available_nodes))
 
         # Try to guess the number of tasks now
-        available_nodes = self.filter_nodes(available_nodes,
-                                            self.sched_access + self.options)
+        # Append `options` to a list containing the elements of `sched_access`
+        # whis is immutable.
+        available_nodes = self.filter_nodes(
+            available_nodes, list(self.sched_access) + self.options)
```

Comment:
It's better if the `SequenceView` supports the `+` operator and returns a new list. The introduction of `SequenceView` must be transparent to the rest of the code.

Reply:
Should addition of two `SequenceView` instances be supported?